openamp/libmetal.cmake

```diff
@@ -60,16 +60,33 @@ else()
   set(LIBMETAL_ARCH ${CONFIG_ARCH})
 endif()
 
-set(PROJECT_VERSION_MAJOR 0)
-set(PROJECT_VERSION_MINOR 1)
-set(PROJECT_VERSION_PATCH 0)
-set(PROJECT_VERSION 0.1.0)
-set(PROJECT_SYSTEM nuttx)
-set(PROJECT_PROCESSOR ${LIBMETAL_ARCH})
-set(PROJECT_MACHINE ${CONFIG_ARCH_CHIP})
-set(PROJECT_SYSTEM_UPPER nuttx)
-set(PROJECT_PROCESSOR_UPPER ${LIBMETAL_ARCH})
-set(PROJECT_MACHINE_UPPER ${CONFIG_ARCH_CHIP})
+# cmake-format: off
+function(libmetal_hdrs_sedexp input_header output_header)
+  execute_process(
+    COMMAND
+      sed
```

Comment:
Do not use unix style commands in cmake！You need to consider cross host platform compatibility when making any build modifications！

Reply:
openamp/libmetal is also based on the CMake rule, could we re-implement the build rule using external project?

Reply:
> Do not use unix style commands in cmake！You need to consider cross host platform compatibility when making any build modifications！

Yes you are right, I just picked it without considering this.
There is internally already has bugfix for removing unix-specific in CMake, I'll upstreamed it

Reply:
> openamp/libmetal is also based on the CMake rule, could we re-implement the build rule using external project?

sure, let me try it using nuttx_add_external_library